File: source-connector/src/main/java/sourceconnector/domain/factory/JSONLogFactory.java
package sourceconnector.domain.factory;

import sourceconnector.domain.log.JSONLog;
import sourceconnector.domain.log.Log;
import sourceconnector.domain.log.LogMetadata;

public class JSONLogFactory implements LogFactory {
  @Override
  public Log create(String payload, LogMetadata logMetadata) {
    return new JSONLog(payload, logMetadata);
  }
}

File: source-connector/src/main/kotlin/sourceconnector/domain/log/factory/JSONLogFactory.kt
package sourceconnector.domain.log.factory

import sourceconnector.domain.log.JSONLog
import sourceconnector.domain.log.Log
import sourceconnector.domain.log.LogMetadata

class JSONLogFactory : LogFactory {
  override fun create(payload: String, logMetadata: LogMetadata): Log {
    return JSONLog(payload, logMetadata)
  }
}
